cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowToken.java

```diff
@@ -92,13 +105,30 @@ void mergeToken(BasicWorkflowToken other) {
 
   @Override
   public void put(String key, String value) {
-    if (nodeName == null) {
-      throw new IllegalStateException("Node name cannot be null.");
+    put(key, Value.of(value));
+  }
+
+  @Override
+  public void put(String key, Value value) {
+    put(key, value, Scope.USER);
+  }
+
+  void put(String key, Value value, Scope scope) {
+    Preconditions.checkNotNull(key, "Null key cannot be added in the WorkflowToken.");
+    Preconditions.checkNotNull(value, String.format("Null value provided for the key '%s'.", key));
+    Preconditions.checkNotNull(value.toString(), String.format("Null value provided for the key '%s'.", key));
```

Comment:
Ok. I think its ok for ``Value`` to continue to accept ``null``, but for ``WorkflowToken`` to make sure its non-null for now.